ibis/expr/types/relations.py

```diff
@@ -778,7 +819,7 @@ def schema(self) -> sch.Schema:
 
     def group_by(
         self,
-        by: str | ir.Value | Iterable[str] | Iterable[ir.Value] | None = None,
+        by: str | ir.Value | Iterable[str] | Iterable[ir.Value] = (),
```

Comment:
Changing the default is fine, but getting rid of `None` support is a user facing change. What if someone is calling `group_by(by=None, metrics=...)`? That's a legitimate call. Why break it?

Reply:
Ohhh, this is from the pre-experimentation phase. Going to restore the `None` support but keep the new default. BTW `None` is a valid value, it gets coerced to a null literal.

Reply:
>  BTW None is a valid value, it gets coerced to a null literal.

That's a breaking change I think, because previously it was being used as a sentinel to mean "no grouping keys"

Reply:
Restored, added a test too.